### PR TITLE
Yoke PID

### PR DESCRIPTION
**Pull request by NotSimon5673**
This is a proposed fix for #43

### CODE REVIEW

File: .SysId/sysid-window.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{
```
Comment (KenwoodFox) [inline .SysId/sysid-window.json]: Dont include this file.

File: .SysId/sysid.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (KenwoodFox) [inline .SysId/sysid.json]: Dont include this file

**Review by KenwoodFox — CHANGES_REQUESTED** [review]

File: rio/subsystems/yoke.py [hunk 3]
```diff
@@ -40,11 +40,17 @@ def __init__(self) -> None:
             CANSparkMaxLowLevel.MotorType.kBrushless,
         )
 
+        # MOVE ME
+        self.starShooter.setInverted(True)
+
```
Comment (KenwoodFox) [inline rio/subsystems/yoke.py]: Why are these here? We already confirmed motor direction for the starboard side, only the port is inverted?

File: rio/subsystems/yoke.py [hunk 4]
```diff
@@ -154,18 +157,22 @@ def setSpeed(self, speed):
         control required.
         """
 
-        self.portShooter.set(speed)
-        self.starShooter.set(-speed)
+        self.portShooter.set(-speed)
+        self.starShooter.set(speed)
```
Comment (KenwoodFox) [inline rio/subsystems/yoke.py]: ??

File: rio/subsystems/yoke.py [hunk 5]
```diff
@@ -154,18 +157,22 @@ def setSpeed(self, speed):
...
 
-    def setVelocity(self, velocity):
```
Comment (KenwoodFox) [inline rio/subsystems/yoke.py:147]: What? We need velocity in a `setVelocity()` method

File: rio/subsystems/yoke.py [hunk 6]
```diff
@@ -154,18 +157,22 @@ def setSpeed(self, speed):
...
 
         # TODO: These need to be inverted, DONT do this here, do this in init
-        self.starPID.setReference(velocity, CANSparkMaxLowLevel.ControlType.kVelocity)
-        self.portPID.setReference(velocity, CANSparkMaxLowLevel.ControlType.kVelocity)
+        self.starPID.setReference(
+            self.setVelocity(), CANSparkMaxLowLevel.ControlType.kVelocity
+        )  # self.setVelocity() sets the velocity feel free, to fix it
+        self.portPID.setReference(
+            self.setVelocity(), CANSparkMaxLowLevel.ControlType.kVelocity
+        )
```
Comment (KenwoodFox) [inline rio/subsystems/yoke.py]: image

**Review by KenwoodFox — CHANGES_REQUESTED** [review]
This is awful, sorry, we will talk tomorrow.

File: rio/subsystems/yoke.py (hunk 3 above)
Comment (NotSimon5673) [reply rio/subsystems/yoke.py]: both of them are set to be inverted they're just set in different places below the PID that is being inverted

Comment (KenwoodFox) [reply rio/subsystems/yoke.py]: Only one motor should be inverted. The other must go the other way, you should not have a negative speed setting further down.

**Review by KhanSimeoni — APPROVED** [review]
Looks kinda sloppy, maybe like a sloppy shooter???

**Review by KenwoodFox — CHANGES_REQUESTED** [review]
no omg XD

**Comment by KenwoodFox** [comment]
If this gets merged it will be merged with #56
